include/flecs/flecs.hpp

```diff
@@ -2855,7 +2863,7 @@ class component_info final {
         s_name = ecs_get_fullpath(world, entity);
     }
 
-    static entity_t id_no_lifecycle(world_t *world = nullptr, const char *name = nullptr) {
+    static entity_t id_no_lifecycle(world_t *world, const char *name = nullptr) {
```

Comment:
The `world = nullptr` shouldn't be removed. It is possible for an application to request an id without providing the world (for example, using `flecs::type_id<T>()`).

Reply:
Reverted